nemo/collections/nlp/models/dialogue/intent_bio_type_classification_model.py

```diff
@@ -0,0 +1,793 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pickle
+from typing import Dict, List, Optional
+
+import torch
+from omegaconf import DictConfig, OmegaConf
+from pytorch_lightning import Trainer
+from torch.utils.data import DataLoader
+
+from nemo.collections.common.losses import AggregatorLoss, CrossEntropyLoss
+from nemo.collections.common.parts import MultiLayerPerceptron
+from nemo.collections.nlp.data.dialogue.data_processor.assistant_data_processor import DialogueAssistantDataProcessor
+from nemo.collections.nlp.data.dialogue.dataset.dialogue_bert_dataset import DialogueBERTDataset
+from nemo.collections.nlp.data.intent_slot_classification import IntentSlotDataDesc, IntentSlotInferenceDataset
+from nemo.collections.nlp.metrics.classification_report import ClassificationReport
+from nemo.collections.nlp.metrics.dialogue_metrics import DialogueClassificationMetrics
+from nemo.collections.nlp.models.nlp_model import NLPModel
+from nemo.collections.nlp.modules.common import SequenceTokenClassifier
+from nemo.collections.nlp.parts.utils_funcs import tensor2list
+from nemo.core.classes import typecheck
+from nemo.core.classes.common import PretrainedModelInfo
+from nemo.utils import logging
+
+UNIQUE_COUNTER = 0
+UNIQUE_EPOCH_COUNTER = 0
+class IntentBIOTypeClassificationModel(NLPModel):
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+        """ Initializes BERT Joint Intent and Slot model.
+        """
+        self.max_seq_length = cfg.dataset.max_seq_length
+        self.cfg = cfg
+        # Check the presence of data_dir.
+        if not cfg.dataset.data_dir or not os.path.exists(cfg.dataset.data_dir):
+            # Set default values of data_desc.
+            self._set_defaults_data_desc(cfg)
+        else:
+            self.data_dir = cfg.dataset.data_dir
+            # Update configuration of data_desc.
+            self._set_data_desc_to_cfg(cfg, cfg.dataset.data_dir, cfg.train_ds, cfg.validation_ds)
+        # init superclass
+        super().__init__(cfg=cfg, trainer=trainer)
+
+        # Initialize Classifier.
+        self._reconfigure_classifier()
+        
+        # Initialize MultiLayerPerceptron
+        self.slot_mlp = MultiLayerPerceptron(
```

Comment:
The naming of this might be better - something like self.bio_slot_mlp

Reply:
This is for slot classification.